### PR TITLE
Support ulimits on Swarm services.

### PR DESCRIPTION
**Pull request by akerouanton**
**- What I did**

1. Bump swarmkit to [d6592ddefd8a5319aadff74c558b816b1a0b2590](https://github.com/docker/swarmkit/commit/d6592ddefd8a5319aadff74c558b816b1a0b2590) to include docker/swarmkit#2967 ;
2. Ulimits field has been added to the ContainerSpec API type, such that the Docker daemon API can now take ulimits when declaring new services, updating or inspecting them ;
3. The cluster executor has been changed to convert from/to Swarmkit type ;

This is related to #40639.

**- How I did it**

**- How to verify it**

I tested my changes by updating docker/cli and using `docker service create|update|inspect` and `docker stack deploy` (I have to create that PR).

**- Description for the changelog**

Support ulimits on Swarm services.

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by thaJeztah** [comment]
https://github.com/moby/moby/pull/41282 was merged; this needs a rebase now

**Comment by cpuguy83** [comment]
Can you squash the last two commits?

**Comment by akerouanton** [comment]
I rebased and squashed my commits :wink:

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Review by thaJeztah — APPROVED** [review]
LGTM

@cpuguy83 double check; this is not something we want to be living in `Resources.Limits` ?

**Comment by thaJeztah** [comment]
Discussing with @cpuguy83 and we should be ok merging as-is; we can update if we think it has to move

**Comment by cpuguy83** [comment]
tldr; didn't want to hold this up and require the contributor to change, we can fix it if necessary. CapAdd/Drop would follow this same line of thinking as well.
